latex/expressingParallelism.tex

```diff
@@ -2249,6 +2249,66 @@ \subsubsection{Defining kernels using OpenCL C kernel objects}
 \codeinline{cl_kernel} is no longer needed in the calling context.
 
 
+%*******************************************************************************
+% OpenCL Kernel Conventions and SYCL
+%*******************************************************************************
+\subsection{OpenCL Kernel Conventions and SYCL}
+\label{sec:opencl_kern_conventions_and_sycl}
+
+OpenCL and SYCL use opposite conventions for the unit stride dimension.  SYCL
+aligns with C++ conventions, which is important to understand from a performance
+perspective when porting code to SYCL.
+
+When specifying a \codeinline{range} as the global or local size
+in a \codeinline{parallel_for} that invokes an OpenCL interop kernel (through
+\codeinline{cl_kernel} interop or \codeinline{compile_with_source}/
+\codeinline{build_with_source}),
+the highest dimension of the range in SYCL will map to the
+lowest dimension within the OpenCL kernel.  That statement applies to both
+an underlying enqueue operation such as \codeinline{clEnqueueNDRangeKernel}
+in OpenCL, and also ID and size queries within the OpenCL kernel.
+For example, a global range specified in SYCL as:
+
+\begin{lstlisting}[style=nonumbers]
+range<3> R{r0,r1,r2};
+\end{lstlisting}
+
+maps to an \codeinline{clEnqueueNDRangeKernel} \codeinline{global_work_size} argument
+of:
+
+\begin{lstlisting}[style=nonumbers]
+size_t cl_interop_range[3] = {r2,r1,r0};
+\end{lstlisting}
+
+Work-item and work-group ID and range queries have the same reversed convention for unit
```

Comment:
The way latex generates the following tables means the two text sections appear first and then the two tables show together, so it reads a bit strange. If we use links to the tables instead I think this will avoid any confusion.

Reply:
Good point.  I switched to a table reference instead.